src/cider/nrepl/middleware/out.clj

```diff
@@ -0,0 +1,87 @@
+(ns cider.nrepl.middleware.out
+  "Change *out* to print on sessions in addition to process out.
+
+  Automatically changes the root binding of *out* to print to any
+  active sessions. An active session is one that has sent at least one
+  \"eval\" op.
+
+  We use an eval message, instead of the clone op, because there's no
+  guarantee that the channel that sent the clone message will properly
+  handle output replies."
+  (:require [cider.nrepl.middleware.util.cljs :as cljs]
+            [clojure.tools.nrepl.middleware :refer [set-descriptor!]]
+            [clojure.tools.nrepl.middleware.interruptible-eval :as ie]
+            [clojure.tools.nrepl.middleware.session :as session])
+  (:import [java.io PrintWriter Writer]))
+
+;;; OutStream
+(defonce original-out *out*)
+
+(declare tracked-sessions-map)
+
+(defmacro do-with-out-messages
+  "Run body with v bound to the output stream of each msg in msg-seq.
+  Also run body with v bound to `original-out`."
+  [[v msg-seq] & body]
+  `(do (let [^Writer ~v original-out]
+         ~@body)
+       (doseq [{:keys [~'session] :as ~'msg} ~msg-seq]
+         (let [^Writer ~v (get @~'session #'*out*)]
+           (try (binding [ie/*msg* ~'msg]
+                  ~@body)
+                ;; If a channel is faulty, dissoc it.
+                (catch Exception ~'e
+                  (swap! tracked-sessions-map dissoc
+                         (:id (meta ~'session)))))))))
+
+(defn fork-out
+  "Returns a PrintWriter suitable for binding as *out* or *err*. All
+  operations are forwarded to all output bindings in the sessions of
+  messages in addition to the server's usual PrintWriter (saved in
+  `original-out`)."
+  [messages]
+  (PrintWriter. (proxy [Writer] []
+                  (close [] (.flush ^Writer this))
+                  (write [& [x ^Integer off ^Integer len]]
```

Comment:
This `&`  + destructuring seems extraneous and I think is responsible for a stacktrace. Well, removing & and the destructuring makes things work :-) :

```
ERROR [2015-09-18 22:03:06,794] com.yammer.dropwizard.jersey.LoggingExceptionMapper: Error handling a request: 8cf72e1a7bae9e6a
! java.lang.IllegalArgumentException: No matching method found: write for class java.io.OutputStreamWriter
! at clojure.lang.Reflector.invokeMatchingMethod(Reflector.java:80)
! at clojure.lang.Reflector.invokeInstanceMethod(Reflector.java:28)
! at cider.nrepl.middleware.out$fork_out$fn__20648.doInvoke(out.clj:45) ~[na:na]
! at clojure.lang.RestFn.invoke(RestFn.java:423)
! at cider.nrepl.middleware.out.proxy$java.io.Writer$ff19274a.write(Unknown Source) ~[na:na]
! at java.io.PrintWriter.write(PrintWriter.java:405) ~[na:1.8.0_45-internal]
! at java.io.PrintWriter.append(PrintWriter.java:1063) ~[na:1.8.0_45-internal]
! at java.io.PrintWriter.append(PrintWriter.java:56) ~[na:1.8.0_45-internal]
! at clojure.core$pr.doInvoke(core.clj:3563)
! at clojure.lang.RestFn.applyTo(RestFn.java:139)
! at clojure.core$apply.invoke(core.clj:630)
! at clojure.core$prn.doInvoke(core.clj:3593)
! at clojure.lang.RestFn.applyTo(RestFn.java:137)
! at clojure.core$apply.invoke(core.clj:630)
```

I'm using this impl of fork-out:
```
(defn fork-out
  "Returns a PrintWriter suitable for binding as *out* or *err*. All
  operations are forwarded to all output bindings in the sessions of
  messages in addition to the server's usual PrintWriter (saved in
  `original-out`)."
  [messages]
  (PrintWriter. (proxy [Writer] []
                  (close [] (.flush ^Writer this))
                  (write
                    ([c]
                     (do-with-out-messages [out messages] (.write out c)))
                    ([s ^Integer off ^Integer len]
                     (do-with-out-messages [out messages] (.write out s off len))))
                  (flush []
                    (do-with-out-messages [out messages] (.flush out))))
                true))
```